pyomo/repn/ampl.py

```diff
@@ -0,0 +1,1261 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import ctypes
+import math
+import operator
+
+from collections import deque
+from operator import itemgetter
+
+from pyomo.common.deprecation import deprecation_warning
+from pyomo.common.errors import DeveloperError, InfeasibleConstraintException, MouseTrap
+from pyomo.common.numeric_types import (
+    native_complex_types,
+    native_numeric_types,
+    native_types,
+    value,
+)
+
+
+from pyomo.core.base import Expression
+from pyomo.core.expr import (
+    NegationExpression,
+    ProductExpression,
+    DivisionExpression,
+    PowExpression,
+    AbsExpression,
+    UnaryFunctionExpression,
+    MonomialTermExpression,
+    LinearExpression,
+    SumExpression,
+    EqualityExpression,
+    InequalityExpression,
+    RangedExpression,
+    Expr_ifExpression,
+    ExternalFunctionExpression,
+)
+from pyomo.core.expr.visitor import StreamBasedExpressionVisitor, _EvaluationVisitor
+from pyomo.repn.util import (
+    BeforeChildDispatcher,
+    ExitNodeDispatcher,
+    ExprType,
+    InvalidNumber,
+    apply_node_operation,
+    complex_number_error,
+    nan,
+    sum_like_expression_types,
+)
+
+
+_CONSTANT = ExprType.CONSTANT
+_MONOMIAL = ExprType.MONOMIAL
+_GENERAL = ExprType.GENERAL
+
+# Feasibility tolerance for trivial (fixed) constraints
+TOL = 1e-8
+
+
+def _create_strict_inequality_map(vars_):
+    vars_['strict_inequality_map'] = {
+        True: vars_['less_than'],
+        False: vars_['less_equal'],
+        (True, True): (vars_['less_than'], vars_['less_than']),
+        (True, False): (vars_['less_than'], vars_['less_equal']),
+        (False, True): (vars_['less_equal'], vars_['less_than']),
+        (False, False): (vars_['less_equal'], vars_['less_equal']),
+    }
+
+
+class text_nl_debug_template(object):
```

Comment:
FYI - don't need `(object)`.

Reply:
Also this class name violates our own standards. I get that this is a legacy thing, but we may want to find places where we violate our own rules at some point.